### PR TITLE
Add 4 new Menu Entries to HMI display for "User Program"

### PR DESCRIPTION
**Pull request by sumedhreddy90**
## Description
This PR enables the users to launch their turtlebot4 application from HMI display by pressing "User Program 1" on HMI menu. Here we have created 4 new publishers that publish boolean values. Upon user selecting "User Program 1" button, we are publishing a bool  flag to be as True. This way user's can use this topic to subscribe in their projects as per their need and application.

today
presentation_1

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

This has been test by running:
```
ros2 topic echo /user_program_1
ros2 topic echo /user_program_2
ros2 topic echo /user_program_3
ros2 topic echo /user_program_4

```
Upon user selecting the desired User Program, the data for the corresponding topic changes to true

Screenshot from 2022-08-16 20-59-52

```bash
# Run this command
ros2 launch turtlebot4_ignition_bringup ignition.launch.py 
```

## Checklist

- [x] I have performed a self-review of my own code and reviewed by @kscottz 
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: turtlebot4_node/include/turtlebot4_node/turtlebot4.hpp [hunk 1]
```diff
@@ -163,13 +171,18 @@ class Turtlebot4 : public rclcpp::Node
...
   rclcpp::TimerBase::SharedPtr power_off_timer_;
+  rclcpp::TimerBase::SharedPtr user_program_timer_;
 
   // Subscribers
   rclcpp::Subscription<sensor_msgs::msg::BatteryState>::SharedPtr battery_sub_;
   rclcpp::Subscription<irobot_create_msgs::msg::WheelStatus>::SharedPtr wheel_status_sub_;
 
   // Publishers
   rclcpp::Publisher<std_msgs::msg::String>::SharedPtr ip_pub_;
+  rclcpp::Publisher<std_msgs::msg::Bool>::SharedPtr user_program_pub_one;
```
Comment (kscottz) [inline turtlebot4_node/include/turtlebot4_node/turtlebot4.hpp:182]: Can we add a comment here to just to make it clear what these topics are used for.

File: turtlebot4_node/src/turtlebot4.cpp [hunk 2]
```diff
@@ -564,6 +604,50 @@ void Turtlebot4::help_function_callback()
   }
 }
 
+/**
+ * @brief User Program 1 callback function to publish bool
```
Comment (kscottz) [inline turtlebot4_node/src/turtlebot4.cpp:608]: Can we make these functions description a bit a clearer please? My understanding is that these callbacks are called when the user selects the menu. I would say something like, "Issues a "True" message when the user selects the program one menu item."

For callbacks I generally like to name after the event that causes them, e.g. "user_program_one_selected_callback", or "power_button_down_callback." The reason you do this is that you often have sets of callbacks for different parts of the pressing a button (e.g. hover, down, up, and pressed).

**Review by kscottz — COMMENTED** [review]
I'll leave @roni-kreinin with final decision on this PR. @sumedhreddy90 if you have the time between now and your last day I would suggest changing the function name and comments to be a bit more clear. 

@roni-kreinin just for some context here, we wanted a dead-simple way for students to load / run their own program on the TB4 without requiring modification of the TB4 node packages. We decided to use this approach to avoid bogging down the TB4 node with services / actions dependent on student code. This approach also avoids requiring the students to install / build the TB4 node from source. Obviously this isn't the best approach, and we wouldn't recommended it for production, but it should provide a good stepping stone for students who are just starting out and want to build simple behaviors on their TB4.  The plan is to have another template node, in a different package, listen for the button state change and dispatch student programs. 

Let me know if this works for you. Happy to discuss.

**Comment by roni-kreinin** [comment]
@kscottz @sumedhreddy90 I would like to propose a different approach to this.

Currently `turtlebot4_node` has a set of predefined functions that can be added to the menu. If the user were to add a function in the config file that is not one of the predefined functions, it will still show up on the menu but selecting it will call `unused_function_callback()` and nothing will happen. 

We can create a publisher that will publish the name of the function when it is called. This way, external nodes will be notified when any function is called, and the name of the function is not restricted to the predefined functions.

For example, the users can set the menu to this
```yaml
menu:
      entries: ["Dock", "Undock", "Program1", "Program2", "Help"]
```
and when they select `Program1`, the publisher will simply publish the name "Program1". The user can have a second node that subscribes to this topic and runs program 1 when it receives this.

Let me know what you think. I can make a PR for this if you would like.

**Comment by roni-kreinin** [comment]
Heres a demo of what I meant. The `Program1` and `Program2` functions don't actually do anything in the node.
function_calls

**Comment by kscottz** [comment]
Oh wow, that's an even simpler solution. I like it.

What do you say @sumedhreddy90 want to give it a shot?

**Comment by roni-kreinin** [comment]
I pushed the code to the `roni-kreinin/functions` branch if you want to test it.

**Comment by roni-kreinin** [comment]
Created new PR #36 to replace this one.

**Pull request closed without merging**